keras_cv/layers/object_detection/retina_net_label_encoder.py

```diff
@@ -137,28 +135,41 @@ def _encode_sample(self, gt_boxes, anchor_boxes):
             label,
         )
 
-        n_boxes = tf.shape(gt_boxes)[0]
+        batch_size = gt_boxes.get_shape().as_list()[0]
+        n_boxes = tf.shape(gt_boxes)[1]
         box_ids = tf.range(n_boxes, dtype=matched_gt_idx.dtype)
-        matched_ids = tf.expand_dims(matched_gt_idx, axis=1)
+        matched_ids = tf.expand_dims(matched_gt_idx, axis=-1)
         matches = box_ids == matched_ids
-        matches = tf.math.reduce_any(matches, axis=0)
+        matches = tf.math.reduce_any(matches, axis=1)
         self.matched_boxes_metric.update_state(
-            tf.zeros((n_boxes,), dtype=tf.int32),
+            tf.zeros(
+                (
+                    batch_size,
+                    n_boxes,
+                ),
+                dtype=tf.int32,
+            ),
             tf.cast(matches, tf.int32),
         )
         return label
 
-    def call(self, images, target_boxes):
-        """Creates box and classification targets for a batch"""
+    def call(self, images, gt_boxes, gt_classes):
+        """Creates box and classification targets for a batch
+
+        Args:
+          images: a batched [batch_size, H, W, C] image float `tf.Tensor`.
+          gt_boxes: a batched [batch_size, num_objects, 4] or ragged batch float ground truth boxes in `bounding_box_format`.
```

Comment:
What is the reason for the gt_ prefix here?

Reply:
Either way is fine with me, seems the consensus here is to use `gt` prefix

Reply:
Gotcha -- I think personally I prefer omitting the prefix, but if this is the consensus among the group then sgtm!